File: C_programs/src/Program_2.c
# include <Program_2.h>

int find_max_diff(int arr[], const int n) {
    int maxDiff = 0;

    for (int i = 0; i < n; ++i) {
        for (int j = n - 1; j > i; --j) {
            if (arr[j] > arr[i] && maxDiff < (j - i))
                maxDiff = j - i;
        }
    }

    return maxDiff;
}

File: Java_Programs/src/com/example/programs/Program_2.java
package com.example.programs;

public class Program_2 {
    public static int findMaxDiff(int[] arr) {
        int maxDiff = 0;
        for (int i = 0; i < arr.length; i++) {
            for (int j = arr.length - 1; j > i; j--) {
                if (arr[j] > arr[i] && maxDiff < (j - i)) {
                    maxDiff = j - i;
                }
            }
        }
        return maxDiff;
    }
}
